doc/file-locking.md

```diff
@@ -0,0 +1,366 @@
+disable schemes as they exist in late 2022. The current library versions at
+this time are 1.10.9, 1.12.3, and 1.13.2. File locks are not present in HDF5
+1.8. The lock feature matrix later in this document will describe the
+limitations of earlier versions.
+
+### Configure option
+
+You can set the file locking defaults at configure time. This sets the defaults
+for the associated properties in the fapl. Users can override the configure
+defaults using `H5Pset_file_locking()` or the `HDF5_USE_FILE_LOCKING`
+environment variable.
+
+- Autotools
+
+    `--enable-file-locking=(yes|no|best-effort)` sets the file locking behavior.
+    `on` and `off` should be self-explanatory. `best-effort` turns file locking
+    on but ignores file locks when they are disabled (default: `best-effort`).
+
+- CMake
+
+    - set `IGNORE_DISABLED_FILE_LOCK` to `ON` to ignore file locks when they
+      are disabled on the file system (default: `ON`).
+    - set `HDF5_USE_FILE_LOCKING` to `OFF` to disable file locks (default: `ON`)
+
+### `H5Pset_file_locking()`
+
+This API call can be used to override the configure defaults. It takes
+`hbool_t` parameters for both the file locking and "ignore file locks when
+disabled on the file system" parameters. The values set here can be
+overridden by the file locking environment variable.
+
+There is a corresponding `H5Pget_file_locking()` call that can be used to check
+the currently set values of both properties in the fapl. **NOTE** that this
+call just checks the property list values. It does **NOT** check the
+environment variables!
+
+### Environment variables
+
+The `HDF5_USE_FILE_LOCKING` environment variable overrides all other file
+locking settings.
+
+HDF5 1.10.0
+- No file locking environment variable
+
+HDF5 1.10.1 - 1.10.6, 1.12.0:
+- `FALSE` turns file locking off
+- Anything else turns file locking on
+- Neither of these values ignores disabled file locks
+- Environment variable parsed at file create/open time
+
+HDF5 1.10.7+, 1.12.1+, 1.13.x:
+- `FALSE` or `0` disables file locking
+- `TRUE` or `1` enables file locking
+- `BEST_EFFORT` enables file locking and ignores disabled file locks
+- Anything else gives you the defaults
+- Environment variable parsed at library startup
+
+### Lock disable scheme interactions
+
+As mentioned above and reiterated here:
+- Configure-time settings set fapl defaults
+- `H5Pset_file_locking()` overrides configure-time defaults
+- The environment variable setting overrides all
+
+If you want to check that file locking is on, you'll need to check the fapl
+setting AND check the environment variable, which can override the fapl.
+
+**!!! WARNING !!!**
+
+Disabling the file locks is at your own risk. If more than one writer process
+modifies an HDF5 file at the same time, the file could be corrupted. If a
+reader process reads a file that is being modified by a writer, the writer
+process might attempt to read garbage and encounter errors or even crash.
+
```

Comment:
It seems the condition above could satisfy those in lines 256 and 257.  What's the difference?